swatch-contracts/src/main/java/com/redhat/swatch/contract/service/ContractService.java

```diff
@@ -200,7 +200,7 @@ public void deleteContract(String uuid) {
   public StatusResponse createPartnerContract(PartnerEntitlementContract contract) {
     StatusResponse statusResponse = new StatusResponse();
 
-    if (!validPartnerEntitlementContract(contract)) {
+    if (!validPartnerEntitlementContract(contract) && !validAzureEntitlementContract(contract)) {
```

Comment:
What is the difference between a PartnerEntitlementContract & an AzureEntitlementContract?

Reply:
The AzureEntitlementContract has `offerID`  which is equal to aws  `venderProductCode` and the `resourceId` is a specific field in azure that is part of the SaaS Purchase, the validation is checking for specific fields that is populated for `aws` and not `azure`

Reply:
So, the "validPartnerEntitlementContract()" is aws specific? If so, that method should probably be renamed or have an aws specific method and an azure specific method that is then called from validPartnerEntitlementContract().